src/adapter/redstone/RedstoneCoreOracle.sol

```diff
@@ -123,9 +145,27 @@ contract RedstoneCoreOracle is PrimaryProdDataServiceConsumerBase, BaseAdapter {
     function _getQuote(uint256 inAmount, address _base, address _quote) internal view override returns (uint256) {
         bool inverse = ScaleUtils.getDirectionOrRevert(_base, base, _quote, quote);
 
-        uint256 staleness = block.timestamp - cacheUpdatedAt;
-        if (staleness > maxCacheStaleness) revert Errors.PriceOracle_TooStale(staleness, maxCacheStaleness);
+        Cache memory _cache = cache;
 
-        return ScaleUtils.calcOutAmount(inAmount, cachedPrice, scale, inverse);
+        if (block.timestamp > _cache.priceTimestamp) {
+            // No need to check price timestamps in the future as they can only get more recent with time.
+            uint256 priceStaleness = block.timestamp - _cache.priceTimestamp;
+            if (priceStaleness > maxStaleness) {
+                revert Errors.PriceOracle_TooStale(priceStaleness, maxStaleness);
+            }
+        }
+        // Reject `getQuote` before first update.
+        if (_cache.price == 0) revert Errors.PriceOracle_InvalidAnswer();
```

Comment:
does it make sense to do this check every time the price is fetched if it's only to prevent fetching price before the first update? I think it'd better to simply leave the `priceTimestamp` as 0 in the constructor so that before the first update we'd revert with `PriceOracle_TooStale`. note that you're already checking `price == 0` in the `updatePrice` function so removing this would be safe

Reply:
true, fixed